Callalen/HikingChallenge.java

```diff
@@ -25,9 +25,11 @@ public static void main(String[] args) throws IOException {
                     grid[i][j].getNeighbors(grid);
                 }
             }
-            Node start = grid[Integer.parseInt(first.substring(2, first.indexOf(",")))][Integer.parseInt(first.substring(first.indexOf(",")+1, first.indexOf(")")))];
-            Node end = grid[Integer.parseInt(first.substring(first.lastIndexOf("(")+1, first.lastIndexOf(",")))][Integer.parseInt(first.substring(first.lastIndexOf(",")+1, first.lastIndexOf(")")))];
-            start.g = 0;
+
+            // use split...your i and j represent ROWS then COLUMNS...the input is COLUMNS then ROWS
+            Node start = grid[Integer.parseInt(first.substring(first.indexOf(",")+1, first.indexOf(")")))][Integer.parseInt(first.substring(2, first.indexOf(",")))];
```

Comment:
If you use split, it makes reading the code MUCH easier!

String[] coords = first.split("[{(),} ]+"); // gives string array with " " at index 0!
Node start = grid[Integer.parseInt(coords[2])][Integer.parseInt(coords[1])];
Node end = grid[Integer.parseInt(coords[4])][Integer.parseInt(coords[3])];